dao/application_authentication_dao.go

```diff
@@ -37,6 +38,25 @@ type applicationAuthenticationDaoImpl struct {
 	UserID   *int64
 }
 
+func (a applicationAuthenticationDaoImpl) getDb() *gorm.DB {
+	if a.TenantID == nil {
+		panic("nil tenant found in applicationAuthentication db DAO")
+	}
+
+	query := DB.Debug()
+	query = query.Where("tenant_id = ?", a.TenantID)
+
+	if a.UserID != nil {
+		query = query.Where("user_id IS NULL OR user_id = ?", a.UserID)
```

Comment:
The `IS NULL` bit maybe is a leftover? Because I think that would match the following:

| id | tenant_id | app_id | auth_id | user_id |
|----|-----------|--------|---------|---------|
| 1  | 12345     | 999    | 888     | 101010    |

Even if the specified user ID is different from `101010`.